src/panels/lovelace/common/create-badge-element.ts

```diff
@@ -0,0 +1,73 @@
+import deepClone from "deep-clone-simple";
+
+import "../badges/hui-entity-filter-badge";
+import "../badges/hui-state-label-badge";
+
+import {
+  createErrorBadgeElement,
+  createErrorBadgeConfig,
+  HuiErrorBadge,
+} from "../badges/hui-error-badge";
+import { LovelaceBadge } from "../types";
+import { LovelaceBadgeConfig } from "../../../data/lovelace";
+
+const BADGE_TYPES = new Set(["entity-filter", "error", "state-label"]);
+const CUSTOM_TYPE_PREFIX = "custom:";
+const TIMEOUT = 2000;
+
+const _createElement = (
+  tag: string,
+  config: LovelaceBadgeConfig
+): LovelaceBadge | HuiErrorBadge => {
+  const element = document.createElement(tag) as LovelaceBadge;
+  try {
+    element.setConfig(deepClone(config));
+  } catch (err) {
+    // tslint:disable-next-line
+    console.error(tag, err);
+    return _createErrorElement(err.message);
+  }
+  return element;
+};
+
+const _createErrorElement = (error: string): HuiErrorBadge =>
+  createErrorBadgeElement(createErrorBadgeConfig(error));
+
+export const createBadgeElement = (
+  config: LovelaceBadgeConfig
+): LovelaceBadge | HuiErrorBadge => {
```

Comment:
Isn't `HuiErrorBadge` part of `LovelaceBadge` ?

Reply:
Copied code from create-card. I didn't question it.